lib/libswan/DBG_dump.c

```diff
@@ -64,4 +66,24 @@ void DBG_dump(const char *label, const void *p, size_t len)
 		passert(sp <= str + elemsof(str));
 		DBG_log("%-*s   %s", (int)sizeof(hex)-1, hex, str);
 	} while (len != 0);
+
+	if (label == NULL) {
+		return;
+	}
+
+	const char *ch = p;
+	for (size_t i = 0; i < init_len; i++) {
+		if (*ch == '\0') {
+			break;
+		}
+		if ((*ch < ' ') || (*ch > '~')) {
+			return;
+		}
+		++ch;
+	}
+
+	char* buf = alloc_bytes(init_len + 1, "dump buf");
```

Comment:
Maybe create a chunk_t with ptr and len, then use DBG_dump_hunk()  so that there is no need for alloc/free ?